ntbs-service/Models/Validations/ValidationAttributes.cs

```diff
@@ -1,20 +1,36 @@
 using System;
-using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
-using System.Reflection;
-using ntbs_service.Models;
-using ntbs_service.Models.Validations;
 
 namespace ntbs_service.Models.Validations
 {
-    public class ValidDateAttribute : RangeAttribute
+    public class ValidDateRangeAttribute : ValidationAttribute
     {
-        public ValidDateAttribute(string startDateString) : base(typeof(DateTime),
-            startDateString, DateTime.Now.ToShortDateString()) {}
+        private readonly DateTime StartDate;
+        public ValidDateRangeAttribute(string startDate) : base()
+        {
+            StartDate = DateTime.Parse(startDate);
+        }
+
+        protected override ValidationResult IsValid(object value, ValidationContext validationContext)
+        {
+            if (value != null)
+            {
+                var date = (DateTime)value;
+                if (date < StartDate)
+                {
+                    return new ValidationResult(ErrorMessage);
+                }
+                if (date > DateTime.Today)
+                {
+                    return new ValidationResult(ValidationMessages.TodayOrEarlier(validationContext.DisplayName));
+                }
+            }
+            return null;
+        }
 
         public override string FormatErrorMessage(string name)

```

Comment:
Is FormatErrorMessage run via `ValidationResult(ErrorMessage)`?

Reply:
yup! :)